### PR TITLE
PRJ-778 Optimize Dockerfile and its workaround

### PR DESCRIPTION
**Pull request by nitrotol**
Resolve [PRJ-778](https://youtrack.jetbrains.com/issue/PRJ-778)

Optimize Dockerfile and its workaround to almost perfect base image

### CODE REVIEW

File: build-tools/install-additional-software.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+# Uncomment this in case uou need extra debug info
+#set -e # Any command which returns non-zero exit code will cause this shell script to exit immediately
+#set -x # Activate debugging to show execution details: all commands will be printed before execution
+
+DOWNLOAD_URL=$1
+# Define filter function
+function string_filter {
+  sed -z 's/  */ /g;s/\n//g' $1
+}
```
Comment (Avvessalom) [inline build-tools/install-additional-software.sh:27]:
```suggestion
string_filter() {
  sed -z 's/  */ /g;s/\n//g' $1
}
```

File: build-tools/install-additional-software.sh [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+# Block for PhpStorm
+  *PhpStorm*)
+    SOFTWARE_LIST=$(string_filter <<- EOM
+
+EOM
+)
+    ADDITIONAL_COMMANDS=$(string_filter <<- EOM
+
+EOM
+    );;
```
Comment (Avvessalom) [inline build-tools/install-additional-software.sh:82]: A lot of duplication, and I have no idea how to make it more compact. If you have, please do

**Review by Avvessalom — COMMENTED** [review]
Thank you for contributing! A quick note on the code style

File: build-tools/install-additional-software.sh (hunk 2 above)
Comment (nitrotol) [reply build-tools/install-additional-software.sh:82]: I have some ideas about this. In few days I`ll try to add smth

Comment (nitrotol) [reply build-tools/install-additional-software.sh:82]: @Avvessalom unfortunately I cant optimize this(

For now I have only one solution without code duplcation, but it required a many (at least one for each variant of IDE) files, witch store software list for each IDE.
But I think this is not a good solution. 

Maybe we can accept this solution and, in future, give more convinient way?

Comment (SerVB) [reply build-tools/install-additional-software.sh:82]: Discussed here: https://github.com/JetBrains/projector-docker/pull/16#discussion_r754174125. So let's leave it if we can't improve it more.

**Comment by SerVB** [comment]
Could you please clarify why there are two PRs (this one and #16) with some colliding changes? Which one you want to merge?

Please also squash some commits it the both PRs: for now, there are many.

**Comment by yole** [comment]
We're discontinuing the development of projector-docker and no longer accepting contributions, sorry.

**Comment by cqupt-zjs** [comment]
您好，俺已经收到了恁里邮件，谢谢哈~

**Pull request closed without merging**